common/src/main/java/org/opensearch/ml/common/dataset/TextSimilarityInputDataSet.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.ml.common.dataset;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.ml.common.annotation.InputDataSet;
+
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.experimental.FieldDefaults;
+
+@Getter
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@InputDataSet(MLInputDataType.TEXT_SIMILARITY)
+public class TextSimilarityInputDataSet extends MLInputDataset {
+    
+   List<String> textDocs;
+
+   String queryText;
+
+    @Builder(toBuilder = true)
+    public TextSimilarityInputDataSet(String queryText, List<String> textDocs) {
+        super(MLInputDataType.TEXT_SIMILARITY);
+        Objects.requireNonNull(textDocs);
+        Objects.requireNonNull(queryText);
+        if(textDocs.isEmpty()) {
+            throw new IllegalArgumentException("pairs must be nonempty");
```

Comment:
[nit] can we add some more user-friendly exception message?

Reply:
ack